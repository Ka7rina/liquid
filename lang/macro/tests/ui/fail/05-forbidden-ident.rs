use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        fn __liquid_noop(&self) {}
    }
}

fn main() {}
