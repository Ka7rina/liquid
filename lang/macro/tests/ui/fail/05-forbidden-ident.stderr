error: identifiers starting with `__liquid` are forbidden in liquid contract
  --> $DIR/05-forbidden-ident.rs:12:12
   |
12 |         fn __liquid_noop(&self) {}
   |            ^^^^^^^^^^^^^
